### PR TITLE
BSA snapshot functions

### PR DESCRIPTION
**Pull request by ChristopherMayes**
Added high level functions to fetch BSA data from archived HDF5 files. 

This finds the appropriate HDF5 file and extracts the data nearest the given timestamp:
```python
from lcls_live.bsa import bsa_snapshot
bsa_snapshot('2021-12-11T00:00:00-08:00', 'cu_hxr')
```

### CODE REVIEW

**Review by jacquelinegarrahan — APPROVED** [review]
LGTM

File: lcls_live/bsa.py [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+
+
+
+def bsa_snapshot(timestamp, beampath, pvnames=None):
+    """
+    Extract as a snapshot (PV values) nearest a timestamp from a BSA HDF5 file.
+    
+    Parameters
+    ----------
+    h5file: str
```
Comment (mattgibbs) [inline lcls_live/bsa.py]: There's no 'h5file' parameter in this function.

**Review by mattgibbs — COMMENTED** [review]
Pretty nice!  A future improvement might be to allow the user to input start and end timestamps, and get all values between those times.  I found one issue with a docstring, other than that, this looks good to go.

File: lcls_live/bsa.py (hunk 1 above)
Comment (ChristopherMayes) [reply lcls_live/bsa.py]: fixed

Comment (ChristopherMayes) [reply lcls_live/bsa.py]: @mattgibbs thanks for catching that!

**Pull request merged**